cookbooks/aws-parallelcluster-install/recipes/efs.rb

```diff
@@ -17,37 +17,107 @@
 
 package_name = "amazon-efs-utils"
 efs_utils_tarball = "#{node['cluster']['sources_dir']}/efs-utils-#{node['cluster']['efs_utils']['version']}.tar.gz"
-return unless platform?('amazon')
+stunnel_tarball = "#{node['cluster']['sources_dir']}/stunnel-#{node['cluster']['stunnel']['version']}.tar.gz"
 
 # Do not install efs-utils if a same or newer version is already installed.
 return if Gem::Version.new(get_package_version(package_name)) >= Gem::Version.new(node['cluster']['efs_utils']['version'])
 
-# Get EFS Utils tarball
-remote_file efs_utils_tarball do
-  source node['cluster']['efs_utils']['url']
-  mode '0644'
-  retries 3
-  retry_delay 5
-  not_if { ::File.exist?(efs_utils_tarball) }
-end
+if platform?('amazon')
+  # On Amazon Linux 2, amazon-efs-utils and stunnel are installed from OS repo.
+  package 'amazon-efs-utils' do
+    retries 3
+    retry_delay 5
+  end
+else
+  # On Ubuntu and CentOS, amazon-efs-utils and stunnel are installed from source
+  # Because their OS repos do not have amazon-efs-utils and new stunnel
+  # Get EFS Utils tarball
+  remote_file efs_utils_tarball do
+    source node['cluster']['efs_utils']['url']
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(efs_utils_tarball) }
+  end
```

Comment:
We duplicate the logic for installing a package from our S3 repo in many places in our cookbook.
What about putting this logic into an helper function and reuse that one?

This may be considered an improvement out of scope for this PR, but let's keep track of it.